### ISSUE TITLE
Sequential and Isolated

### ISSUE THREAD

**Issue by nargov** [etorreborre/specs2#366]
I have a test that's been passing for a long time, which uses both `sequential` and `isolated`.
We've been trying to upgrade to version 3.5, but then this test stopped passing.
It seems that when `sequential` is added, `isolated` stops working.

I can round up a failing test for this if you'd like.

**Comment by etorreborre** [comment]
Yes a small test case is always useful, thanks.

**Comment by etorreborre** [comment]
FYI I tried to reproduce on 2 simple specs2 and I couldn't find any issue
```scala
class TestSpec extends Specification with ScalaCheck { def is = sequential ^ isolated ^ s2"""
 test1 $e1
 test2 $e2

"""
  def e1 = { val b = i; i += 1; "executing test1".pp; b === 0}
  def e2 = { val b = i; i += 1; "executing test2".pp; b === 0}

  var i = 0
}

class TestMutableSpec extends mutable.Specification {
  sequential
  isolated

  "test1" >> e1
  "test2" >> e2

  def e1 = { val b = i; i += 1; "executing test1".pp; b === 0}
  def e2 = { val b = i; i += 1; "executing test2".pp; b === 0}

  var i = 0
}
```

**Comment by nargov** [comment]
I have a reproducing test, but it's quite large, I'll try to remove the uninteresting parts and see if it still breaks.

**Comment by nargov** [comment]
Okay, here is my problem.
I have a trait that I use to integrate JMock with specs2.
Because of the nature of the Mockery, I need a new one for each example. So, what I did was this:

```trait MyTrait extends ArgumentsShortcuts with ArgumentsArgs{
  isolated
}
class TestMutableSpec extends Specification {
  sequential

  "test1" >> e1
  "test2" >> e2

  def e1 = { val b = i; i += 1; "executing test1".pp; b === 0}
  def e2 = { val b = i; i += 1; "executing test2".pp; b === 0}

  var i = 0
}```

In this case, isolated is ignored. 
However, in an older version (3.3 I think) it still worked.
So, am I going about it wrong, or is there an issue?

**Comment by etorreborre** [comment]
Thanks for the example. I have found where the problem is but I need more time to find the proper solution and understand why the behaviour changed.

In the meantime a workaround is to use ` args(sequential = true, isolated = true)` as one declaration.

**Comment by etorreborre** [comment]
I fixed in in `3.5-20150429115212-9152a24`. Can you please grab that version and tell me if it works for you?

**Comment by nargov** [comment]
Confirmed, it works.

**Comment by nargov** [comment]
I'm guessing this will be released in 3.5.1, do you have any idea when that is due?

**Comment by etorreborre** [comment]
I'll try to release it this week-end.
